src/DynamicMvvm.Uno.WinUI/Extensions/DispatcherQueueExtensions.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/main/License.md
+// See reference: https://github.com/CommunityToolkit/WindowsCommunityToolkit/blob/main/Microsoft.Toolkit.Uwp/Extensions/DispatcherQueueExtensions.cs
+
+using System;
+using System.Threading.Tasks;
+using Windows.Foundation.Metadata;
+using Microsoft.UI.Dispatching;
+using System.Runtime.CompilerServices;
+
+namespace MessageDialogService.Extensions
+{
+	internal static class DispatcherQueueExtensions
+	{
+		/// <summary>
+		/// Indicates whether or not <see cref="DispatcherQueue.HasThreadAccess"/> is available.
+		/// </summary>
+		private static readonly bool IsHasThreadAccessPropertyAvailable = ApiInformation.IsMethodPresent("Windows.System.DispatcherQueue", "HasThreadAccess");
+
+		/// <summary>
+		/// Invokes a given function on the target <see cref="DispatcherQueue"/> and returns a
+		/// <see cref="Task"/> that completes when the invocation of the function is completed.
+		/// </summary>
+		/// <param name="dispatcher">The target <see cref="DispatcherQueue"/> to invoke the code on.</param>
+		/// <param name="function">The <see cref="Action"/> to invoke.</param>
+		/// <param name="priority">The priority level for the function to invoke.</param>
+		/// <returns>A <see cref="Task"/> that completes when the invocation of <paramref name="function"/> is over.</returns>
+		/// <remarks>If the current thread has access to <paramref name="dispatcher"/>, <paramref name="function"/> will be invoked directly.</remarks>
+		public static Task EnqueueAsync(this DispatcherQueue dispatcher, Action function, DispatcherQueuePriority priority = DispatcherQueuePriority.Normal)
```

Comment:
I think we should rename this to RunAsync, or anything that's not "enqueue" because it doesn't sound like it's waiting for the execution.

Reply:
I'll rename it. I think they originally named it EnqueueAsync since DispatcherQueue uses TryEnqueue instead of RunAsync for CoreDispatcher. This extension is there to make sure it's async.